test_app/tests/jwt_consumer/common/test_auth.py

```diff
@@ -476,9 +477,14 @@ def test_correctly_authenticate_if_the_cached_key_is_invalid_but_the_new_key_is_
             with mock.patch('ansible_base.jwt_consumer.common.auth.JWTCert.get_decryption_key', create_mock_method(jwt_cert_field_changes)):
                 request = mocked_http.mocked_parse_jwt_token_get_request('with_headers')
                 jwt_auth = JWTAuthentication()
-                user = django_user_model.objects.create_user(username=jwt_token.unencrypted_token['user_data']['username'], password="password")
+                if not new_user:
+                    user = django_user_model.objects.create_user(username=jwt_token.unencrypted_token['user_data']['username'], password="password")
                 created_user, _ = jwt_auth.authenticate(request)
-                assert created_user == user
+                if not new_user:
+                    assert created_user == user
+                else:
+                    assert created_user.username == jwt_token.unencrypted_token['user_data']['username']
+                    assert str(created_user.resource.ansible_id) == jwt_token.unencrypted_token['sub']
```

Comment:
To clarify, this was failing before this patch, the ansible_id values were not the same. This is the source of a conflict in what I was trying to do.